### ISSUE TITLE
How do I build typesrcipt for vitejs react?

### ISSUE THREAD

**Issue by TMinh25** [GetStream/stream-chat-react#1447]
- "stream-chat": "^6.1.0",
- "stream-chat-react": "^7.0.0",

When I run "yarn build" it returns alot of typing errors in react. How do I build in vite

![image](https://user-images.githubusercontent.com/58514384/158054453-3ceb6298-e633-488c-94e6-10e4c862a934.png)

**Comment by TMinh25** [comment]
even the example for vite in this repo cannot build with typescript

**Comment by petyosi** [comment]
@TMinh25 sorry for the trouble, `stream-chat` 6.0 was recently refactored to a single generic, and we have not published the corresponding update in `stream-chat-react`. Use `stream-chat: 5.6.0` instead.

**Comment by TMinh25** [comment]
> @TMinh25 sorry for the trouble, `stream-chat` 6.0 was recently refactored to a single generic, and we have not published the corresponding update in `stream-chat-react`. Use `stream-chat: 5.6.0` instead.

does `stream-chat: 5.6.0` compatible with `stream-chat-react: 7.0.0` or we have to downgrade the `stream-chat-react` too?

**Comment by petyosi** [comment]
> does stream-chat: 5.6.0 compatible with stream-chat-react: 7.0.0 or we have to downgrade the stream-chat-react too?

the two versions should work together.

**Comment by TMinh25** [comment]
Okay I've downgrade the version:

- `stream-chat: 5.0.0`
- `stream-chat-react: 6.13.0`

it solved almost all typing errors
how ever there's still 1 error from `stream-chat-react` with `VirtuosoProps`: this is the stack

```
node_modules/stream-chat-react/dist/components/MessageList/VirtualizedMessageList.d.ts:9:31 - error TS2314: Generic type 'VirtuosoProps<D, C>' requires 2 type argument(s).

9     additionalVirtuosoProps?: VirtuosoProps<UnknownType>;
                                ~~~~~~~~~~~~~~~~~~~~~~~~~~
```

any thought on this?

**Comment by petyosi** [comment]
You need to fix the version of `react-virtuoso` to `2.6.0`, before the additional context generic.

**Comment by TMinh25** [comment]
> You need to fix the version of `react-virtuoso` to `2.6.0`, before the additional context generic.

Can you explain the part _before the additional context generic_?

**Comment by petyosi** [comment]
`v2.7.0` of the `react-virtuoso` package introduced an additional generic, which `stream-chat-react` has not introduced yet. So you need to fix the version of the package to `2.6.0`.

**Comment by TMinh25** [comment]
> `v2.7.0` of the `react-virtuoso` package introduced an additional generic, which `stream-chat-react` has not introduced yet. So you need to fix the version of the package to `2.6.0`.

Do I fix the version in my package.json or somewhere else? Because it doesn't work if I change in my project's package.json

**Comment by TMinh25** [comment]
could you help me please? I have to finish build my project in 2 days
